test/functional/change_stream_tests.js

```diff
@@ -518,20 +530,24 @@ describe('Change Streams', function() {
             assert.equal(change.operationType, 'invalidate');
 
             // now expect the server to close the stream
-            changeStream.once('close', done);
+            changeStream.once('end', done);
```

Comment:
hmm, I think this is indeed supposed to be listening to `close`, `end` is the `ReadableStream` event but we want to verify that the server explicitly closes this cursor.

Reply:
I'm not seeing the server closing the cursor. `Cursor.prototype.close` is never called. What is the mechanism by which we expect the server to close the cursor?

Reply:
For instance here: https://github.com/mongodb/node-mongodb-native/blob/3.0.0/lib/cursor.js#L1177.   When the stream attempts to issue a getMore and it has an empty batch result, or the cursor id is set to 0, then the cursor calls `this/self.close()`. 

I do recall some issues here, which might have been why I had this commented out code left in: https://github.com/mongodb/node-mongodb-native/blob/3.0.0/lib/cursor.js#L1201-L1208, give it a shot with that reenabled. This might require more involved pairing.

Reply:
Uncommenting that code does indeed make the tests pass. Unfortunately, it also manages to break 14 other tests. I will investigate more.